Dockerfile

```diff
@@ -1,93 +1,131 @@
+# to ensure that the app is only started once the DB container
+# is ready
+COPY ./wait-for-it.sh /app/
 
-# Set the working dir and copy the app
-WORKDIR /app
-COPY --chown=geoadmin:geoadmin .env.default /
-COPY --chown=geoadmin:geoadmin ./spec /spec/
-COPY --chown=geoadmin:geoadmin ./app /app/
+# Overwrite the version.py from source with the actual version
+ARG VERSION=unknown
+RUN echo "APP_VERSION = '$VERSION'" > app/config/version.py
 
 ARG GIT_HASH=unknown
 ARG GIT_BRANCH=unknown
 ARG GIT_DIRTY=""
 ARG AUTHOR=unknown
-ARG VERSION=unknown
-ARG TARGET=
 LABEL git.hash=$GIT_HASH
 LABEL git.branch=$GIT_BRANCH
 LABEL git.dirty="$GIT_DIRTY"
 LABEL author=$AUTHOR
 LABEL version=$VERSION
 
-# Overwrite the version.py from source with the actual version
-RUN echo "APP_VERSION = '$VERSION'" > /app/config/version.py
+WORKDIR ${INSTALL_DIR}/app/
+USER ${USER}
 
 ###########################################################
-# Container to perform tests/management/dev tasks
-FROM base as debug
-
-LABEL target=debug
-
-RUN cd /tmp && \
-    pipenv install --system --deploy --ignore-pipfile --dev
-
-# this is only used with the docker-compose setup within CI
-# to ensure that the app is only started once the DB container
-# is ready
-COPY ./wait-for-it.sh /app/
-
-# for testing/management, settings.py just imports settings_dev
-RUN echo "from .settings_dev import *" > /app/config/settings.py \
-    && chown geoadmin:geoadmin /app/config/settings.py
-
-# Collect static files, uses the .env.default settings to avoid django raising settings error
-RUN APP_ENV=default LOGGING_CFG=0 ./manage.py collectstatic --noinput
+# Builder container
+FROM base as builder
+RUN apt-get update -qq \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y gdal-bin \
+    # dev dependencies
+    binutils libproj-dev \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* \
+    && pip3 install pipenv \
+    && pipenv --version
 
-USER geoadmin
+ENV PIPENV_VENV_IN_PROJECT=1
+COPY Pipfile.lock Pipfile ./
+RUN pipenv sync
 
-EXPOSE $HTTP_PORT
+COPY --chown=${USER}:${GROUP} .env.default ./
+COPY --chown=${USER}:${GROUP} spec/ spec/
+COPY --chown=${USER}:${GROUP} app/ app/
 
-# entrypoint is the manage command
-ENTRYPOINT ["python3"]
+# on prod, settings.py needs to be replaced to import settings_prod instead of settings_dev
+RUN echo "from .settings_prod import *" > app/config/settings.py \
+    && chown ${USER}:${GROUP} app/config/settings.py
 
 
 ###########################################################
 # Container to use in production
 FROM base as production
-
 LABEL target=production
 
-# on prod, settings.py just import settings_prod
-RUN echo "from .settings_prod import *" > /app/config/settings.py \
-    && chown geoadmin:geoadmin /app/config/settings.py
+RUN apt-get update -qq \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y gdal-bin \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
```

Comment:
This layer could be moved in the base image as it used on all images.

Reply:
The problem is then that if you change the list of debug tools of dev dependencies to install, you could then land with a mismatch of versions (an old version of gdal-bin and new versions of the other dependencies) thats why I thought that it's better to install everything that comes from apt in a single step.

Reply:
Ok